lib/milestone-db/memory.js

```diff
@@ -23,7 +25,9 @@ function MemoryMilestoneDB(options) {
 MemoryMilestoneDB.prototype = Object.create(MilestoneDB.prototype);
 
 MemoryMilestoneDB.prototype.getMilestoneSnapshot = function(collection, id, version, callback) {
-  if (!this._isValidVersion(version)) return process.nextTick(callback, new ShareDBError(4001, 'Invalid version'));
+  if (!this._isValidVersion(version)) {
+    return process.nextTick(callback, new ShareDBError(ERROR_CODE.ERR_UNKNOWN_ERROR, 'Invalid version'));
```

Comment:
We shouldn't use unknown errors explicitly - let's use something like `ERR_MILESTONE_ARGUMENT_INVALID`